src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -125,6 +125,11 @@ private async Task<IActionResult> GetResultAsync(HttpContext context, IFunctionE
                     var applicationLifetime = context.RequestServices.GetService<IApplicationLifetime>();
                     CancellationToken cancellationToken = applicationLifetime != null ? applicationLifetime.ApplicationStopping : CancellationToken.None;
                     await functionExecution.ExecuteAsync(context.Request, cancellationToken);
+
+                    if (context.Items.TryGetValue(ScriptConstants.AzureFunctionsDuplicateHttpHeadersKey, out object value))
```

Comment:
This simple check doesn't need to be a first class middleware component, with the overhead associated with that.

Reply:
Would be nice to sill maintain this logic outside of this middleware, though. How big of an impact was this having?

Reply:
Not a big impact really - this was just about reducing the number of pieces of middleware we register in the pipeline. For simple post invocation things like this, why not just include them in the invoke middleware?

Looking at performance profiles, the we dispatch through a large number of middleware components, each involving async state transitions/overhead. Granted, some early experiments I've done with middleware compression didn't yield large improvements, but its an area I want to investigate more.

Reply:
Makes sense, but I just want us to be careful about separation of concerns and making sure we don't end up with a bloated piece of middleware that does more than it should, is difficult to maintain and ends up being touched more than it should as that would lead to regressions. We should be able to optimize existing pieces of middleware to ensure we're applying good patterns to avoid going async when not needed, but in many cases, even if there's a small gain from combining logic, if that leads to code that is more prone to issues, I think we should keep them separate.

Having them as separate pieces also enables us to more easily test the middleware surface, which also gives us another layer of safety.